ui_framework/src/global_styling/mixins/_namespace.scss

```diff
@@ -0,0 +1,99 @@
+// Namespace mixin
+// This forces component, child, modifier and state namespacing.
+// Check the styleguide for more detail.
+
+$childSeparator: '__';
+$modifierSeparator: '--';
+$stateSeparator: '.is';
+
+// Is this mixin itself nested within a mixin?
+@function isNested($selector) {
+    $selector: selectorToString($selector);
+    @if str-index($selector, $modifierSeparator) {
+        @return true;
+    } @else if str-index($selector, $stateSeparator){
+      @return true;
+    } @else if str-index($selector, $childSeparator){
+      @return true;
+    } @else {
+      @return false;
+    }
+}
```

Comment:
Instead of a series of "else ifs" which return something, can we use the pattern of "early exits", with a default return value at the end? It's a bit easier to read:

```scss
// Is this mixin itself nested within a mixin?
@function isNested($selector) {
  $selector: selectorToString($selector);
  @if str-index($selector, $modifierSeparator) {
      @return true;
  }

  @if str-index($selector, $stateSeparator){
    @return true;
  }

  @if str-index($selector, $childSeparator){
    @return true;
  }

  @return false;
}
```